### PR TITLE
ansible: update node from 14 to 16

### PR DESCRIPTION
**Pull request by MoLow**
Fixes: https://github.com/nodejs/build/issues/3329

### CODE REVIEW

File: ansible/roles/jenkins-workspace/tasks/main.yml [hunk 1]
```diff
@@ -129,11 +129,16 @@
...
 
 - name: Add nodesource repo
+  apt_repository:
+    repo: deb https://deb.nodesource.com/node_18.x {{ release_codename.stdout }} main
+    state: present
+  
+- name: Remove nodesource 14 repos
   apt_repository:
     repo: deb https://deb.nodesource.com/node_14.x {{ release_codename.stdout }} main
     state: present
```
Comment (nschonni) [inline ansible/roles/jenkins-workspace/tasks/main.yml:139]:
```suggestion
    state: absent
```

**Comment by MoLow** [comment]
I had to use node 16 and not 18 due to https://github.com/nodesource/distributions/issues/1392#issuecomment-1134101584.
I guess it is time to start upgrading ubuntu 18 to newer versions

**Comment by MoLow** [comment]
I have ran this on `test-equinix-ubuntu2204-x64-1`, `test-equinix-ubuntu2204-x64-2`, `test-ibm-ubuntu1804-x64-1`

https://github.com/nodejs/build/blob/a68447566e5933be814861523dfaa24e3e280ad2/ansible/playbooks/jenkins/worker/create.yml#L56-L67

**Comment by richardlau** [comment]
> I have ran this on `test-equinix-ubuntu2204-x64-1`, `test-equinix-ubuntu2204-x64-2`, `test-ibm-ubuntu1804-x64-1`
> 
> https://github.com/nodejs/build/blob/a68447566e5933be814861523dfaa24e3e280ad2/ansible/playbooks/jenkins/worker/create.yml#L56-L67

The two packetnet servers are gone and can be removed from the above. They were replaced by the equinix servers.

I'm okay with going to Node.js 16, or attempting to upgrade the IBM Cloud machine (test-ibm-ubuntu1804-x64-1) to Ubuntu 22.04 to match the equinix machines.

**Comment by MoLow** [comment]
> The two packetnet servers are gone and can be removed from the above. They were replaced by the equinix servers.

removed: https://github.com/nodejs/build/pull/3331

> I'm okay with going to Node.js 16, or attempting to upgrade the IBM Cloud machine (test-ibm-ubuntu1804-x64-1) to Ubuntu 22.04 to match the equinix machines.

This is probably beyond the scope of this PR (which is already deployed)

**Review by richardlau — APPROVED** [review]

**Pull request merged**